main.cpp

```diff
@@ -96,189 +101,430 @@ inf
+DoubleType& DoubleType::subtract( double rhs)
 {
-    return lhs - rhs;
+    *value -= rhs;
+    return *this;
 }
 
-double DoubleType::multiply(double lhs, double rhs)
+DoubleType& DoubleType::multiply( double rhs)
 {
-    return lhs * rhs;
+    *value *= rhs;
+    return *this;
 }
 
-double DoubleType::divide(double lhs, double rhs)
+DoubleType& DoubleType::divide( double rhs)
 {
     if (rhs == 0.0)
     {
-        std::cout << "warning, floating point division by zero returns 'inf' !" << std::endl;
+        std::cout << "warning: floating point division by zero!"  << std::endl;
     }
-
-    return lhs / rhs;
+    *value /= rhs;
+    return *this;
 }
 
-struct IntType
+//------------------------------------------------------
+DoubleType& DoubleType::add(FloatType& ft)
 {
-    int add(int lhs, int rhs);
-    int subtract(int lhs,int rhs);
-    int multiply(int lhs, int rhs);
-    int divide(int lhs, int rhs);
-};
+    return add(*ft.value);
+}
+DoubleType& DoubleType::subtract(FloatType& ft)
+{
+    return subtract(*ft.value);
+}
+DoubleType& DoubleType::multiply(FloatType& ft)
+{
+    return multiply(*ft.value);
+}
+DoubleType& DoubleType::divide(FloatType& ft)
+{
+    return divide(*ft.value);
+}
+//------------------------------------------------------
+DoubleType& DoubleType::add(DoubleType& dt)
+{
+    return add(*dt.value);
+}
+DoubleType& DoubleType::subtract(DoubleType& dt)
+{
+    return subtract(*dt.value);
+}
+DoubleType& DoubleType::multiply(DoubleType& dt)
+{
+    return multiply(*dt.value);
+}
+DoubleType& DoubleType::divide(DoubleType& dt)
+{
+    return divide(*dt.value);
+}
+//------------------------------------------------------
+DoubleType& DoubleType::add(IntType& it)
+{
+    return add(*it.value);
+}
+DoubleType& DoubleType::subtract(IntType& it)
+{
+    return subtract(*it.value);
+}
+DoubleType& DoubleType::multiply(IntType& it)
+{
+    return multiply(*it.value);
+}
+DoubleType& DoubleType::divide(IntType& it)
+{
+    return divide(*it.value);
+}
+//------------------------------------------------------
 
-int IntType::add(int lhs, int rhs)
+//======================================================
+IntType& IntType::add( int rhs )
 {
-    return lhs + rhs;
+    *value += rhs;
+    return *this;
 }
 
-int IntType::subtract(int lhs,int rhs)
+IntType& IntType::subtract( int rhs )
 {
-    return lhs - rhs;
+    *value -= rhs;
+    return *this;
 }
 
-int IntType::multiply(int lhs, int rhs)
+IntType& IntType::multiply( int rhs )
 {
-    return lhs * rhs;
+    *value *= rhs;
+    return *this;
 }
 
-int IntType::divide(int lhs, int rhs)
+IntType& IntType::divide( int rhs )
 {
     if (rhs == 0)
     {
-        std::cout << "error, integer division by zero will crash the program!" << std::endl << "returning lhs" << std::endl;
-        return lhs;
+        std::cout << "error: integer division by zero is an error and will crash the program!" << std::endl;
+        return *this;
     }
+    *value /= rhs;
+    return *this;
+}
```

Comment:
D.R.Y.
you wrote `return *this;` twice in the function.  Find an alternate logic that will eliminate this code duplication.
- [ ] revise removing the code duplication.